iib/web/models.py

```diff
@@ -752,11 +753,16 @@ def _from_json(
         Architecture.validate_architecture_json(add_arches)
 
         # Validate binary_image is correctly provided
-        binary_image = request_kwargs.pop('binary_image')
-        if not isinstance(binary_image, str):
+        binary_image = request_kwargs.pop('binary_image', None)
+        if binary_image and not isinstance(binary_image, str):
             raise ValidationError('"binary_image" must be a string')
+        elif not binary_image and not current_app.config['IIB_BINARY_IMAGE_CONFIG']:
+            raise ValidationError(
+                '"binary_image" is required when IIB_BINARY_IMAGE_CONFIG is empty'
```

Comment:
This error message is exposing an internal detail to the user. I would just say it is required.
```suggestion
                '"binary_image" is required when IIB_BINARY_IMAGE_CONFIG is empty'
```